src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -69,6 +69,8 @@ public class ConnectionService
         /// </summary>
         internal Dictionary<string, ConnectionInfo> OwnerToConnectionMap { get; } = new Dictionary<string, ConnectionInfo>();
 
+        public Dictionary<string, Boolean> IsConnected = new Dictionary<string, bool>();

```

Comment:
I don't think this is needed based on the code you have, you are adding/removing stuff to it but never uses it.